### ISSUE TITLE
Returning to main screen after saving form data

### ISSUE THREAD

**Issue by AndrewM-** [geopaparazzi/geopaparazzi#561]
Currently Geopaparazzi returns to the main form with the six icons after a custom form data has been saved.  If you are doing a weed survey or some other very repetitive survey, it would be better to return to the list of forms after saving form data.  For helicopter surveys in particular, returning to the list of forms is critical as time is very short and I need to be able to see the form I want and open it instantly.

**Comment by brentfraser** [comment]
Wow, I was just going to file an issue on this.  I think it would make a good Note setting:
On Save return to:
    o Current Form
    o Form List
    o Main View (default)

**Comment by AndrewM-** [comment]
@brentfraser I get reluctant to report all the little bits as one can degenerate into a serial pest if you don't excercise some restraint.  For me Geopaparazzi is the best data collector I have come across with the possible exception of Fulcrum (which I have only seen demonstrated).  However, I built my own system which I call PhotoHarvester which works with a Garmin GPS and ordinary non-GPS cameras and this system is so productive that it takes the edge of me using Geopaparazzi.  The philosophy of PhotoHarvester is to go somewhere and photograph absolutely everything.  300 photos per day in normal and the record is over 3000 (helicopter survey).  I don't enter data on each photo, in fact most of the data is entered at folder level.  PhotoHarvester is also pythonic in approach in that you only do what you have to do now and you address future problems in the future.  In the first instance, I only have to record the subjects in the folder.  If I need to or want to use the data in the folder, then I will come back some time in the future and review the photos and convert them into data.  Not having to process all the data collected is liberating.  I also often have study everything at once.  For a high voltage transmission line, I will fly along in a helicopter and photography about 20 subjects from native veg, farming, soils, surface rock, low voltage power lines, cattle yards and houses which have not been mapped, erosion gullies and so on.  In contrast, Geopaparazzi and similar systems require data entry in the field.  I can imagine using Geopaparazzi for recording trails and minimal form data and also taking photos with the camera app and integrating these with the Gpap data later on my desktop.  I moving in this direction as I need higher levels of performance than I get with using the camera within Geopaparazzi.  There is also an issue with how photos are stored in the gpap file (photo name is put in form name field, when the word 'photo' should be and the name of the photos should be somewhere else).

**Comment by brentfraser** [comment]
Yes, I would say Geopaparazzi is a GPS track recorder first with some note taking and picture storage abilities.  Sounds like your application is more a imagery/video application with some positioning requirements.  I've tried things like OpenStreetCam and 3d model building from aerial imagery.  These things are compute and storage intense, but I think they have great promise.  I'd love to have some rapid imaging (5 seconds? like OpenStreetCam) with voice notes in Geopaparazzi.

**Comment by AndrewM-** [comment]
Yes I agree. Geopaparazzi is also a mobile GIS where I can show overlays as well.  It is the data collection side that is inefficient in most use cases and this applies to all mobile device data collection systems.

**Comment by moovida** [comment]
I am not sure I understand this (referring to the original issue, not the discussion). You are saying that you want to:

* create custom form, push back
* arrive on the form list (the one you get by long tapping on the notes icon in the map view)

instead of:

* create custom form, push back
* arrive on the form buttons view

Right?
I am not sure about the reason. It is because you can reopen the note quickly? To review it? To modify it? Or are you talking about previous notes? 
I am not sure I am getting the point here.

**Comment by brentfraser** [comment]
The problem is I/we want to:
- enter a note, save it
- return to either:
    1. the same form screen, ready to create a new note, or
    2. the screen showing the form buttons [aka sections] ready to select the form
Currently it returns all the way to the main dashboard screen.  To get to 2. above would require only one button press (and 2 presses to get to the desired form), but it can be annoying when you've got many notes to enter.

**Comment by AndrewM-** [comment]
I think the implementation is not quite right.  This is what I would expect.

1. From the dashboard, open forms menu, fill form and save, return to forms menu
2. From map, open forms menu, fill form and save, return to map.

When launching data entry forms from the map, it is better to return directly to the map.
When launching data entry forms from the dashboard, it is better to return the forms menu.

In the field, I mostly launch forms from the dashboard and may not visit the map even once in a full day.
